### PR TITLE
JAMES-2544 Concurrent testing for mail queue - test only

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[95be9d57d7dae000663935c6d633369428a8f595] [ERROR]   RabbitMQClusterTest$ClusterNodesFailure.connectingToAClusterWithAFailedRabbit » Completion
```

test this please

**Comment by chibenwa** [comment]
```
[95be9d57d7dae000663935c6d633369428a8f595] [ERROR]   RabbitMQClusterTest$ClusterSharing.queuesShouldBeDeclarableOnAnotherNode » Completion
```

test this please

**Comment by chibenwa** [comment]
Forced pushed to solve conflict

**Comment by chibenwa** [comment]
```
[8d5f3b94aa68c43c2949c637a65994164ca30191] Location: /org/apache/james/imap/scripts/UidSearchAtomsIndexer.test:2322
[8d5f3b94aa68c43c2949c637a65994164ca30191] LastClientMsg: A203 UID SEARCH FROM "Timothy"
[8d5f3b94aa68c43c2949c637a65994164ca30191] Expected: '\* SEARCH 1'
[8d5f3b94aa68c43c2949c637a65994164ca30191] Actual   : '* SEARCH'
[8d5f3b94aa68c43c2949c637a65994164ca30191] 
[8d5f3b94aa68c43c2949c637a65994164ca30191] [INFO] 
[8d5f3b94aa68c43c2949c637a65994164ca30191] [INFO] Results:
[8d5f3b94aa68c43c2949c637a65994164ca30191] [INFO] 
[8d5f3b94aa68c43c2949c637a65994164ca30191] [ERROR] Errors: 
[8d5f3b94aa68c43c2949c637a65994164ca30191] [ERROR]   ElasticSearchUidSearchOnIndexTest>UidSearchOnIndex.testSearchAtomsUS:52 » InvalidServerResponse
```

test this please

**Comment by chibenwa** [comment]
CI failed due to insufficient disk space

test this please

**Comment by chibenwa** [comment]
```

[8d5f3b94aa68c43c2949c637a65994164ca30191] [ERROR] Failures: 
[8d5f3b94aa68c43c2949c637a65994164ca30191] [ERROR]   ManageableMailQueueContract.clearShouldRemoveAllElements:577 
[8d5f3b94aa68c43c2949c637a65994164ca30191] Expecting empty but was:<[org.apache.james.queue.api.ManageableMailQueue$MailQueueItemView@539c4830]>
```

test this please

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java [hunk 1]
```diff
@@ -343,10 +345,42 @@ default void deQueueShouldWaitForAMailToBeEnqueued(ExecutorService executorServi
...
+            })
+            .threadCount(threadCount)
+            .operationCount(operationCount)
+            .runSuccessfullyWithin(1, TimeUnit.MINUTES);
+
+        assertThat(
+            dequeuedMails.stream()
+                .map(Mail::getName)
+                .distinct())
+            .hasSize(totalDequeuedMessages);
```
Comment (trantienduchn) [inline server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java]: I would like to have another assertion mailqueue now is empty

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java [hunk 2]
```diff
@@ -378,6 +378,43 @@ default void concurrentEnqueueDequeueShouldNotFail() throws Exception {
...
+            })
+            .threadCount(threadCount)
+            .operationCount(operationCount)
+            .runSuccessfullyWithin(Duration.ofMinutes(1));
+
+        assertThat(
+            dequeuedMails.stream()
+                .map(Mail::getName)
+                .distinct())
+            .hasSize(totalDequeuedMessages);
```
Comment (trantienduchn) [inline server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java]: idem

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java (hunk 1 above)
Comment (chibenwa) [reply server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java]: I can not as this is part of the regular mailQueue. I can not access the management capabilities to easily do this change.

We could assert that dequeue timeout but this is a rather complicated assertion, whose intent won't be clear.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**